pydantic/_internal/_generate_schema.py

```diff
@@ -245,68 +268,70 @@ def model_schema(self, cls: type[BaseModel]) -> core_schema.CoreSchema:
         Since models generate schemas for themselves this method is public and can be called
         from within BaseModel's metaclass.
         """
-        model_ref, schema = self._get_or_cache_recursive_ref(cls)
-        if schema is not None:
-            return schema
-
-        fields = cls.model_fields
-        decorators = cls.__pydantic_decorators__
-        check_decorator_fields_exist(
-            chain(
-                decorators.field_validators.values(),
-                decorators.field_serializers.values(),
-                decorators.validators.values(),
-            ),
-            fields.keys(),
-        )
```

Comment:
This diff isn't actually as bad as it seems, it's coming form indenting the body of all of these functions 1 more tab/4 spaces because I added a context manager at the top of every one.